package psmweb;


enum psmType
{
	ProcessCommandRequest = 0;
	ProcessCommandResponse = 1;

	CounterCommandRequest = 2;
	CounterCommandResponse = 3;

	HealthCheck = 8;
	HealthAck = 9;
}

enum ProcessCommandType
{
	START = 0;
	RESTART = 1;
	STOP = 2;
	ADDLIST = 3;
	DELETELIST = 4;
	ALLSTOP = 5;
}

message wsProcessCommandRequest
{
	required int32 token = 1;
	required ProcessCommandType type = 2;
	required string processName = 3;
}

enum Result
{
	SUCCESS = 0;
	FAILURE = 1;
}

message swProcessCommandResponse
{
	required int32 token = 1;
	required ProcessCommandType type = 2;
	required Result result = 3;
	optional string failReason = 4;
}

enum CounterCommandType
{
	CADDLIST = 1;
	CDELETELIST = 2;
}

message wsCounterCommandRequest
{
	required int32 token = 4;
	required CounterCommandType type = 1;
	repeated string countername = 2;
	required bool ismachine = 3;
}


message swCounterCommandResponse
{
	required int32 token = 1;
	required CounterCommandType type = 2;
	required Result result = 3;
	optional string failReason = 4;
}

message wsHealthCheck
{
}

message swHealthAck
{
}